src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -375,7 +375,14 @@ public static bool RoutePartsEqual(object a, object b)
             var sa = a as string;
             var sb = b as string;
 
-            if (sa != null && sb != null)
+            // In case of strings, consider empty and null the same.
```

Comment:
This change is part of the 4 failing Dispatcher tests in MVC. After this change, I was able to bring down functional test failures from 4 to 2. I want to discuss about the remaining 2 functional tests in person as they ideally do not use the same route setup w.r.t dispatcher

Reply:
Hmm this is interesting. I'm surprised to see that we have a bug here. 

Can you provide a little explanation of what exactly happens?

Reply:
Oh nevermind I found the explanation elsewhere 😆